File: entregable 2/murciaSkyline.py
# Creamos una clase para cada Skyline el cual tendrá edificios dentro.
from typing import List


class SkyLine:
    def __init__(self, cantidadEdificios, caso):
        self.edificios = []
        self.caso = caso

    def agregar_edificio(self, edificio):
        self.edificios.append(edificio)

# Creamos la clase edificio para almacenar el alto y el ancho individualmente.
class Edificio:
    def __init__(self, altura, ancho):
        self.altura = altura
        self.ancho = ancho

## fin de las clases ##

## principo del código main ##
def main():
    # La primera línea del input es la cantidad de casos de test
    cantidadCasos = int(input())

    # inicializamos la lista de todos los skylines
    skyLinesTotales = [] #[Skyline1, ..., SkylineN]

    for caso in range(cantidadCasos):
        # La primera línea del input del caso, es la cantidad de edificios en el skyline
        cantidadEdificios = int(input())

        alturas = []
        anchos = []

        # instancia del skyline actual
        skyLinesTotales.append(SkyLine(cantidadEdificios, caso + 1))

        datos_altura_linea = input().split()
        datos_ancho_linea = input().split()

        # Iteración sobre la cantidad de edificios que tenemos
        for j in range(cantidadEdificios):
            alturas.append(int(datos_altura_linea[j]))
            anchos.append(int(datos_ancho_linea[j]))
            edif = Edificio(alturas[j], anchos[j])
            skyLinesTotales[caso].agregar_edificio(edif)

        # esto me va a devolver algo del estilo:
        # skyLinesTotales = [[edificio1, ..., edificioM] = skyLine0, ..., (edificio1, ..., edificioM') = skyLine[#cantidadDeCasos]]
    
    # una vez que terminamos de tomar el input, podemos trabajar con el arreglo de skylines.
    for i in range(cantidadCasos):
        int maxCreciente = maximoCreciente(skyLinesTotales[i])
        int maxDecreciente = maximoDecreciente(skyLinesTotales[i])

        if(maxCreciente > maxDecreciente):
            return 0
        else:
            return 1


def maximoCreciente(Skyline : List[SkyLine]) -> int:
    suma = 0
    for i in range(1, len(SkyLine)):
        for j in range(i):
            if (Skyline[i].altura > Skyline[j].altura):
                suma += Skyline[i].ancho







        

def mayorSubseqCreciente(skyline : List[Edificio]) -> int:
    return 0


def mayorSubseqDecreciente(skyline : List[Edificio]) -> int:
    return 0

File: entregable 2/murciaSkyline.cpp
#include <iostream>
#include <vector>
using namespace std;

class Edificio {
public:
    int altura;
    int ancho;

    Edificio(int alturaEdificio, int anchoEdificio){
        altura = alturaEdificio;
        ancho = anchoEdificio;
    }
};

class Skyline {
public:
    vector<Edificio> edificios;
    int caso;

    Skyline(int cant_edificios, int caso){
        edificios.reserve(cant_edificios);
        this->caso = caso;
    }

    void agregarEdificio(Edificio edificio){
        edificios.push_back(edificio);
    }

    int maxSubsecuenciaCreciente(){
        vector<int> widths(edificios.size(), 0);

        for (int i = 0; i < edificios.size(); ++i) {
            widths[i] = edificios[i].ancho;
        }

        for (int i = 1; i < edificios.size(); ++i) {
            for (int j = 0; j < i; ++j) {
                if(edificios[i].altura > edificios[j].altura && widths[i] < edificios[i].ancho + widths[j]){
                    widths[i] = widths[j] + edificios[i].ancho;
                }
            }
        }

        int maxValue = 0;
        for (int width : widths) {
            if(width > maxValue){
                maxValue = width;
            }
        }
        return maxValue;
    }

    int maxSubsecuenciaDecreciente(){
        vector<int> widths(edificios.size(), 0);

        for (int i = 0; i < edificios.size(); ++i) {
            widths[i] = edificios[i].ancho;
        }

        for (int i = 1; i < edificios.size(); ++i) {
            for (int j = 0; j < i; ++j) {
                if(edificios[i].altura < edificios[j].altura && widths[i] < edificios[i].ancho + widths[j]){
                    widths[i] = widths[j] + edificios[i].ancho;
                }
            }
        }

        int maxValue = 0;
        for (int width : widths) {
            if(width > maxValue){
                maxValue = width;
            }
        }
        return maxValue;
    }

    int crece_(){
        return maxSubsecuenciaCreciente() >= maxSubsecuenciaDecreciente();
    }
};

int main() {
    int cant_casos;
    cin >> cant_casos;

    vector<Skyline> casos_totales;

    for (int i = 0; i < cant_casos; ++i) {
        int cantidad_edificios;
        cin >> cantidad_edificios;

        vector<int> alturas(cantidad_edificios);
        vector<int> anchos(cantidad_edificios);

        for (int j = 0; j < cantidad_edificios; ++j) {
            cin >> alturas[j];
        }

        for (int j = 0; j < cantidad_edificios; ++j) {
            cin >> anchos[j];
        }

        Skyline skyline(cantidad_edificios, i + 1);

        for (int j = 0; j < cantidad_edificios; ++j) {
            Edificio building(alturas[j], anchos[j]);
            skyline.agregarEdificio(building);
        }
        casos_totales.push_back(skyline);
    }

    for (size_t j = 0; j < casos_totales.size(); j++) {
        string result;

        if (casos_totales[j].crece_()) {
            result = "Increasing (" + to_string(casos_totales[j].maxSubsecuenciaCreciente()) + "). " + "Decreasing (" + to_string(
                    casos_totales[j].maxSubsecuenciaDecreciente()) + ").";
        } else {
            result = "Decreasing (" + to_string(casos_totales[j].maxSubsecuenciaDecreciente()) + "). " + "Increasing (" + to_string(
                    casos_totales[j].maxSubsecuenciaCreciente()) + ").";
        }

        cout << "Case " << casos_totales[j].caso << ". " << result << endl;
    }

}
